### PR TITLE
first cut at better build explanation

### PR DESCRIPTION
**Pull request by masinter**
This is a first cut at trying to explain what's going on in all of these scripts.
Not sure how useful this is.

At this level of detail, documentation belongs in the repo rather than the web site.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -1,34 +1,93 @@
...
+The actual building itself is done with various shell scripts, found in the `scripts` directory in the medley repository. Most of the scripts have a minimal sanity check that they are being run from the medley repository.
+
+Historically, building the medley image (called a "sysout") was called "doing a loadup".  Back in the day, a loadup took the better part of a day, and no one would do the whole thing -- there was no automation.
+
+## Make everything
+
+The shell command:
 ```
-./scripts/loadup-and-release.sh
+time ./scripts/loadup-all.sh =apps && time ./scripts/loadup-db.sh
```
Comment (nbriggs) [inline BUILDING.md:30]: `-apps` not `=apps`

**Comment by blakemcbride** [comment]
Thanks for updating the instructions.  While GitHub actions are all fine and dandy, linking Medley too tightly with GitHub will prematurely age Medley due to future GitHub changes (which are going to happen far, far sooner than changes to shell scripts.)

**Comment by masinter** [comment]
Maybe the part about using GitHub Actions belongs in a different document? 

You can *build* Medley without GitHub or even without git installed. 

The GitHub actions are for making a release for distribution.

File: BUILDING.md [hunk 2]
```diff
@@ -1,34 +1,106 @@
...
+The workflow pulls the latest Maiko image from Docker Hub and the Release Assets from the latest Medley release, generally defined as medley-YYMMDD.  The Medley Docker image adds in Tight VNC Server and retrieves the two tarballs associated with a release, one containing the sysouts and the other the other needed files source, fonts, etc.  The contents are uncompressed and loaded into the Medley directory structure.
+-->
+
+# Building Medley for yourself
+
+The actual building itself is done with various shell scripts, found in the `scripts` directory in the medley repository. Most of the scripts have a minimal sanity check that they are being run from the medley repository.
+
+Historically, building the medley image (called a "sysout") was called "doing a loadup".  Back in the day, a loadup took the better part of a day, and no one would do the whole thing -- there was no automation.
+
+## Prerequisites
```
Comment (pamoroso) [inline BUILDING.md:26]: It's worth making explicit a number of implicit assumptions or details:

- Maiko (and `ldeinit`) must be built before building Medley
- to make everything, the NoteCards source tree must be accessible to the build process
- source tree layout, e.g.
```
src
  maiko
  medley
  notecards
```
- from which directory to run which script
- whether to set `MEDLEYDIR` and to what value (`MEDLEYDIR` seems required but I'm not sure about `MAIKODIR`)

File: BUILDING.md [hunk 3]
```diff
@@ -1,34 +1,106 @@
...
+The shell command:
 ```
-./scripts/loadup-and-release.sh
+time ./scripts/loadup-all.sh -apps && time ./scripts/loadup-db.sh
 ```
-to go on to make the tgz files and release them
+does everything; on a fast machine it takes 4-6 minutes, most of which is spent in the `loadup-db.sh` step. Without the `loadup-db.sh` step, it runs in 22 seconds on a fast system.
 
-# Using github actions
+These are explained in reverse order:
```
Comment (pamoroso) [inline BUILDING.md:47]: A few lines above the document advises to run `loadup-all.sh` and `loadup-db.sh` to make everything. So where do the scripts described below fit in? Should the user run them explicitly? Under what circumstances should the scripts be run?

File: BUILDING.md (hunk 2 above)
Comment (masinter) [reply BUILDING.md:26]: I added some things to help address @pamoroso points above. I'm sure more is needed, but could we please get this out?

**Review by nbriggs — APPROVED** [review]
Seems OK as far as it goes.

**Pull request merged**